### ISSUE TITLE
--server now expects a file of some sort

### ISSUE THREAD

**Issue by jtalkington** [httperf/httperf#22]
With the latest checkin, the behavior of --server now expects a file of some sort to get the hostnames from. The file format is not documented, and this breaks automated tools (specifically Autobench.)

I'd suggest reverting the behavior of the --server argument, and adding a new one (e.g. --server-list-file) for the new behavior.

**Comment by cheteen** [comment]
Yeah, I hit the same issue, and found the same thing. Cost me quite a bit of time stracing and trawling through the code to get this far. Looks like inserting a line with IPs into a file kind of works, but it won't recognise hostnames. Also, even if you get this, the second request on the same connection comes with an extra line, so only the first request in a connection works, so the request generation code may be broken as well. I would first clarify the --server list argument situation before looking further into that.

I've personally rolled back to the last 2014 commit (ed5c6311abf10d0ef7357bf9e6e9942c1fa181fa), and that seems to be working as expected.

All in all, I second the suggestion above adding a new argument and not breaking backwards compatibility. There may be a further defect here with request generation logic. And, we probably need some kind of automated testing here.

**Comment by scippio** [comment]
:+1: 

PS: I don't know how to use this new option :cry: I try:
echo "google.com" > google
httperf --server google

but still I get: httperf.core_addr_intern: invalid server address google.com :cry:

**Comment by scippio** [comment]
:warning: I must remove empty line! ... omg.

**Comment by thaddeusmt** [comment]
Yeah, the `--server` directive doesn't work as described in the documentation right now. Rolling back to an older commit as cheteen mentioned got things working again. Either the documentation or the code could use an update so the match.

**Comment by beedaan** [comment]
I can't get httperf to run at all.  When I try the following command `httperf --server google.com`, the result is
```
httperf --client=0/1 --server=google.com --port=80 --uri=/ --send-buffer=4096 --recv-buffer=16384 --num-conns=1 --num-calls=1
httperf: failed to increase number of open file limit: Invalid argument~/Desktop```

I'm running on Mac 10.10.5.  I also tried creating a file with hosts and passing that in.  I got the same result.  Why is it saying that the current directory is an invalid argument?

**Comment by shivam-maharshi** [comment]
Hi beedaan,
Please use this command if you want to make it work with the latest version. It worked for me.
httperf --server hostname \ --port 80 --uri /test.html \ --rate 150 --num-conn 27000 \ --num-call 1 --timeout 5

**Comment by beedaan** [comment]
@shivam-maharshi, thank you for the reply!  I ran the command you posted and still had issues
```
vagrant@vagrant-ubuntu-trusty-32:~$ httperf --server bheussler.name \ --port 80 --uri /index.htm \ --rate 150 --num-conn 27000 \ --num-call 1 --timeout 5
httperf --timeout=5 --client=0/1 --server=bheussler.name --port=80 --uri=/index.htm --send-buffer=4096 --recv-buffer=16384 --num-conns=27000 --num-calls=1
httperf: can't open bheussler.name
```
I noticed some warnings when I was updating the generated configuration files with `autoreconf -i`
```
...
libtoolize: Consider adding `AC_CONFIG_MACRO_DIR([m4])' to configure.ac and
libtoolize: rerunning libtoolize, to keep the correct libtool macros in-tree.
libtoolize: Consider adding `-I m4' to ACLOCAL_AMFLAGS in Makefile.am.
configure.ac:11: warning: AM_INIT_AUTOMAKE: two- and three-arguments forms are deprecated.  For more info, see:
configure.ac:11: http://www.gnu.org/software/automake/manual/automake.html#Modernize-AM_005fINIT_005fAUTOMAKE-invocation
...
```

**Comment by shivam-maharshi** [comment]
Hi beedan,

I see that the error you received has changed now - httperf: can't open bheussler.name. Can you infer something from this ? On a different note I would suggest that fumbling around randomly would definitely waste your time without getting to near to solving this problem. Here are 2 good references I followed for Ubuntu, I would suggest you remove your httperf completely and then follow the references step by step. Hope that would solve the problem.

Reference -
1. http://www.hpl.hp.com/research/linux/httperf/httperf-man-0.9.txt
2. http://www.hpl.hp.com/research/linux/httperf/wisp98/html/doc003.html

**Comment by bogdanr** [comment]
This is a very significant issue!

**Comment by aliartiza75** [comment]
Why this issue is not resolved yet?

**Comment by aliartiza75** [comment]
I removed the `https://` from the URL and i was able to get it running.

**Comment by erikarn** [comment]
hi! can someone send through a patch? I'd love to merge in a fix!

**Comment by erikarn** [comment]
if people think the best thing to do is make --server do the old behaviour and add a new feature, i can do that too.

**Comment by aliartiza75** [comment]
@erikarn kindly do it.
